src/main/java/com/provectus/oddplatform/puller/task/PullerTask.java

```diff
@@ -13,14 +17,31 @@
     private final AdapterRestService adapterRestService;
     private final PlatformRestService platformRestService;
 
+    private final Generator generator = new Generator();
+
     @Override
     public void run() {
-        log.info("Running puller task for Data Source: {}", dataSourceDto.getOddrn());
+        log.info("Running puller task for Data Source: {}", dataSourceDto.getId());
 
         adapterRestService
-            .getRawEntityList(dataSourceDto.getEndpoint())
+            .getEntities(dataSourceDto.getEndpoint())
+            .flatMap(this::substituteOddrn)
             .flatMap(platformRestService::ingest)
             .doOnError(t -> log.error(t.getMessage()))
-            .subscribe(r -> log.info("Puller task has been completed. Data Source: {}", dataSourceDto.getOddrn()));
+            .subscribe(r -> log.info("Puller task has been completed. Data Source: {}", dataSourceDto.getId()));
+    }
+
+    private Mono<DataEntityList> substituteOddrn(final DataEntityList dataEntityList) {
+        final ODDPlatformDataSourcePath oddrnPath = ODDPlatformDataSourcePath.builder()
```

Comment:
Each time OddrnPath is created. Move out of the run stacktrace or cache

Reply:
may be create it in constructor? It depends only on final field

Reply:
Moved it out of the PullerTask scope